### PR TITLE
Fix the UI theme switch - add to local storage

### LINKED ISSUE #1175: Settings: Theme choice doesn't persist correctly
### Describe the Bug

When you choose an alternate theme, like dark mode, it saves the setting to `localStorage`, but on refresh it will show light mode again. It's not until you go to the settings again that it refreshes the local state.

The solution would be to follow what we are doing for the code editor theme, which is persisted in `localStorage` and also stored in the `settingsStore`. See `SourceCodeEditor.tsx` and `AppearanceModal.tsx` components for reference.

### Steps to Reproduce the Bug or Issue

1. Go to Appearance settings
2. Toggle to Dark Mode for the UI theme
3. Close modal, see it's in dark mode
4. Refresh page, see it's in light mode again
5. Go back to Appearance settings, and it changes back to dark mode

### Screenshots or Videos

Kapture 2023-01-26 at 20 43 13

### Platform

- OS: macOS
- Browser: Chrome
- Version: 0.6.1

### PR DESCRIPTION
**Pull request by tplevko**
Fixed the UI theme switch - works now also in VSCode

closes https://github.com/KaotoIO/kaoto-ui/issues/1175

### CODE REVIEW

File: src/components/KaotoToolbar.tsx [hunk 1]
```diff
@@ -102,6 +104,16 @@ export const KaotoToolbar = ({ toggleCatalog, toggleCodeEditor, hideLeftPanel }:
     });
   }, []);
 
+  // configure UI theme
+  useEffect(() => {
+    const uiTheme = localStorage.getItem(LOCAL_STORAGE_UI_THEME_KEY) ?? "true";
+    if (uiTheme === "true") {
+      htmlTagElement?.classList.remove(THEME_DARK_CLASS);
+    } else {
+      htmlTagElement?.classList.add(THEME_DARK_CLASS);
```
Comment (lordrip) [inline src/components/KaotoToolbar.tsx:113]: It feels strange to attach a class to the root document, I'm wondering what would happen if we embed Kaoto editor inside another application, I guess that the dark theme will be applied to the root as well :thinking: . I couldn't find any related docs from https://www.patternfly.org/

**Review by lordrip — DISMISSED** [review]
LGTM, my only doubt is the class assigned to the root element, but I couldn't find anything on the docs

**Comment by kahboom** [comment]
@apupier would you mind to test this one?

**Comment by kahboom** [comment]
I tested locally for VSCode and it works as expected, though I was surprised it did because on running `yarn build:dev` for VSCode (following [this](https://github.com/KaotoIO/vscode-kaoto/blob/main/CONTRIBUTING.md#how-to-manually-test-latest-kaoto-ui)) there is a relevant warning:

```
WARNING in ../kaoto-ui/dist/lib/store/settingsStore.js 21:104-130
export 'LOCAL_STORAGE_UI_THEME_KEY' (imported as 'LOCAL_STORAGE_UI_THEME_KEY') was not found in 'constants' (possible exports: DH_CHECK_P_NOT_PRIME, DH_CHECK_P_NOT_SAFE_PRIME, DH_NOT_SUITABLE_GENERATOR, DH_UNABLE_TO_CHECK_GENERATOR, E2BIG, EACCES, EADDRINUSE, EADDRNOTAVAIL, EAFNOSUPPORT, EAGAIN, EALREADY, EBADF, EBADMSG, EBUSY, ECANCELED, ECHILD, ECONNABORTED, ECONNREFUSED, ECONNRESET, EDEADLK, EDESTADDRREQ, EDOM, EDQUOT, EEXIST, EFAULT, EFBIG, EHOSTUNREACH, EIDRM, EILSEQ, EINPROGRESS, EINTR, EINVAL, EIO, EISCONN, EISDIR, ELOOP, EMFILE, EMLINK, EMSGSIZE, EMULTIHOP, ENAMETOOLONG, ENETDOWN, ENETRESET, ENETUNREACH, ENFILE, ENGINE_METHOD_ALL, ENGINE_METHOD_CIPHERS, ENGINE_METHOD_DH, ENGINE_METHOD_DIGESTS, ENGINE_METHOD_DSA, ENGINE_METHOD_ECDH, ENGINE_METHOD_ECDSA, ENGINE_METHOD_NONE, ENGINE_METHOD_PKEY_ASN1_METHS, ENGINE_METHOD_PKEY_METHS, ENGINE_METHOD_RAND, ENGINE_METHOD_STORE, ENOBUFS, ENODATA, ENODEV, ENOENT, ENOEXEC, ENOLCK, ENOLINK, ENOMEM, ENOMSG, ENOPROTOOPT, ENOSPC, ENOSR, ENOSTR, ENOSYS, ENOTCONN, ENOTDIR, ENOTEMPTY, ENOTSOCK, ENOTSUP, ENOTTY, ENXIO, EOPNOTSUPP, EOVERFLOW, EPERM, EPIPE, EPROTO, EPROTONOSUPPORT, EPROTOTYPE, ERANGE, EROFS, ESPIPE, ESRCH, ESTALE, ETIME, ETIMEDOUT, ETXTBSY, EWOULDBLOCK, EXDEV, F_OK, NPN_ENABLED, O_APPEND, O_CREAT, O_DIRECTORY, O_EXCL, O_NOCTTY, O_NOFOLLOW, O_NONBLOCK, O_RDONLY, O_RDWR, O_SYMLINK, O_SYNC, O_TRUNC, O_WRONLY, POINT_CONVERSION_COMPRESSED, POINT_CONVERSION_HYBRID, POINT_CONVERSION_UNCOMPRESSED, RSA_NO_PADDING, RSA_PKCS1_OAEP_PADDING, RSA_PKCS1_PADDING, RSA_PKCS1_PSS_PADDING, RSA_SSLV23_PADDING, RSA_X931_PADDING, R_OK, SIGABRT, SIGALRM, SIGBUS, SIGCHLD, SIGCONT, SIGFPE, SIGHUP, SIGILL, SIGINT, SIGIO, SIGIOT, SIGKILL, SIGPIPE, SIGPROF, SIGQUIT, SIGSEGV, SIGSTOP, SIGSYS, SIGTERM, SIGTRAP, SIGTSTP, SIGTTIN, SIGTTOU, SIGURG, SIGUSR1, SIGUSR2, SIGVTALRM, SIGWINCH, SIGXCPU, SIGXFSZ, SSL_OP_ALL, SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION, SSL_OP_CIPHER_SERVER_PREFERENCE, SSL_OP_CISCO_ANYCONNECT, SSL_OP_COOKIE_EXCHANGE, SSL_OP_CRYPTOPRO_TLSEXT_BUG, SSL_OP_DONT_INSERT_EMPTY_FRAGMENTS, SSL_OP_EPHEMERAL_RSA, SSL_OP_LEGACY_SERVER_CONNECT, SSL_OP_MICROSOFT_BIG_SSLV3_BUFFER, SSL_OP_MICROSOFT_SESS_ID_BUG, SSL_OP_MSIE_SSLV2_RSA_PADDING, SSL_OP_NETSCAPE_CA_DN_BUG, SSL_OP_NETSCAPE_CHALLENGE_BUG, SSL_OP_NETSCAPE_DEMO_CIPHER_CHANGE_BUG, SSL_OP_NETSCAPE_REUSE_CIPHER_CHANGE_BUG, SSL_OP_NO_COMPRESSION, SSL_OP_NO_QUERY_MTU, SSL_OP_NO_SESSION_RESUMPTION_ON_RENEGOTIATION, SSL_OP_NO_SSLv2, SSL_OP_NO_SSLv3, SSL_OP_NO_TICKET, SSL_OP_NO_TLSv1, SSL_OP_NO_TLSv1_1, SSL_OP_NO_TLSv1_2, SSL_OP_PKCS1_CHECK_1, SSL_OP_PKCS1_CHECK_2, SSL_OP_SINGLE_DH_USE, SSL_OP_SINGLE_ECDH_USE, SSL_OP_SSLEAY_080_CLIENT_DH_BUG, SSL_OP_SSLREF2_REUSE_CERT_TYPE_BUG, SSL_OP_TLS_BLOCK_PADDING_BUG, SSL_OP_TLS_D5_BUG, SSL_OP_TLS_ROLLBACK_BUG, S_IFBLK, S_IFCHR, S_IFDIR, S_IFIFO, S_IFLNK, S_IFMT, S_IFREG, S_IFSOCK, S_IRGRP, S_IROTH, S_IRUSR, S_IRWXG, S_IRWXO, S_IRWXU, S_IWGRP, S_IWOTH, S_IWUSR, S_IXGRP, S_IXOTH, S_IXUSR, UV_UDP_REUSEADDR, W_OK, X_OK)
 @ ../kaoto-ui/dist/lib/store/index.js 6:0-32 6:0-32
 @ ../kaoto-ui/dist/lib/kogito-integration/KogitoEditorIntegrationProvider.js 4:0-70 16:38-54 17:51-74 18:39-80
 @ ../kaoto-ui/dist/lib/kogito-integration/index.js 4:0-50 4:0-50
 @ ./src/webview/KaotoEditorEnvelopeApp.ts 1:0-81 6:23-41
```

**Review by apupier — CHANGES_REQUESTED** [review]
- if I change to dark mode, then close VS Code, then reopen VS Code, reopen the Kaoto editor, the Kaoto Editor is in Dark mode but the the 3 dots -> Appearance is saying that it is in light mode
image
- The warning mentioned by Rachel in https://github.com/KaotoIO/kaoto-ui/pull/1551#issuecomment-1484934289 is really strange, I think it needs to be looked at.
- Please provide the issue number in the commit
- it is also convenient to provide it in the PR description so that when it is merged, it is closing automatically the issue

File: src/store/settingsStore.tsx [hunk 2]
```diff
@@ -26,6 +26,7 @@ export const initialSettings: ISettings = {
   name: 'integration',
   namespace: '',
   editorIsLightMode: localStorage.getItem(LOCAL_STORAGE_EDITOR_THEME_KEY) === 'true',
+  uiLightMode: localStorage.getItem(LOCAL_STORAGE_UI_THEME_KEY) == null ? true : localStorage.getItem(LOCAL_STORAGE_UI_THEME_KEY) === 'true',
```
[the remark this answers is not in the document]
Comment (tplevko) [reply src/store/settingsStore.tsx]: Thanks for the hint - looks better

File: src/components/KaotoToolbar.tsx (hunk 1 above)
Comment (tplevko) [reply src/components/KaotoToolbar.tsx:113]: I know, I haven't find either any info regarding this. Do you think there is a better place to have this logic?

**Comment by kahboom** [comment]
@tplevko I followed the steps here using your branch for `kaoto-ui`: https://github.com/KaotoIO/vscode-kaoto/blob/main/CONTRIBUTING.md#how-to-manually-test-latest-kaoto-ui

@apupier @lordrip  were you able to reproduce this error? from here: https://github.com/KaotoIO/kaoto-ui/pull/1551#issuecomment-1484934289

From `kaoto-ui` with your branch:
Screen Shot 2023-03-27 at 4 18 52 pm

From `vscode-kaoto` with `main` branch:

Screen Shot 2023-03-27 at 4 20 27 pm
Screen Shot 2023-03-27 at 4 20 42 pm

**Comment by lordrip** [comment]
@kahboom @apupier I'm checking the error at the moment

**Comment by apupier** [comment]
> > it sounds like it is not specific to VS Code. If I refresh page on firefox, I always get the light theme first, and if I go to appearance the dark theme is selected and applied:
> 
> @apupier that's what this issue it's all about, to fix that smile

and based on my gif, I tested with Operate First instead of my local instance built with this PR 🤦

File: src/components/KaotoToolbar.tsx (hunk 1 above)
Comment (tplevko) [reply src/components/KaotoToolbar.tsx:113]: I tried to add it to the main div and didn't work. Seems like the [demo react app ](https://github.com/patternfly/patternfly-react/blob/83c54ab517dd329d924d69ade0430c389aaff5b1/packages/react-integration/demo-app-ts/src/App.tsx#L44)actually does it the same way.

**Comment by kahboom** [comment]
Can confirm that with the latest changes VSCode Kaoto loads without issues and the warning is gone. Only one warning remains but I think this might have been there already previously, maybe @apupier can confirm:

```
WARNING in shared module react
No required version specified and unable to automatically determine one. Unable to find required version for "react" in description file (/Users/kahboom/Projects/vscode-kaoto/node_modules/react/package.json). It need to be in dependencies, devDependencies or peerDependencies.

WARNING in shared module react-dom
No required version specified and unable to automatically determine one. Unable to find required version for "react-dom" in description file (/Users/kahboom/Projects/vscode-kaoto/node_modules/react-dom/package.json). It need to be in dependencies, devDependencies or peerDependencies.
```

**Comment by apupier** [comment]
the twio warnings about shared modules react and react-dom were already there

**Comment by kahboom** [comment]
Per @apupier this PR will have to wait until post 1.0 release, unfortunately, as we don't have sufficient automated testing for VSCode Kaoto in place to merge so close to the release.

**Review by kahboom — DISMISSED** [review]

**Comment by apupier** [comment]
There is a conflict to resolve

**Review by apupier — DISMISSED** [review]
working fine currently in VS Code but it requires to fix the conflict.

I would recommend to rebase on top of main branch to have a clean history instead of several merge commits.

**Comment by tplevko** [comment]
thanks for checking with latest VScode @apupier. Should be rebased, removed the merge commits and conflict is resolved

**Review by kahboom — APPROVED** [review]
thank you @tplevko !

**Review by lordrip — APPROVED** [review]
Tested it with standalone and VSCode, working fine :+1:

**Pull request merged**